### ISSUE TITLE
Error: container has runAsNonRoot and image has non-numeric user (fluent), cannot verify user is non-root

### ISSUE THREAD

**Issue by ag237** [SumoLogic/sumologic-kubernetes-collection#1229]
**Describe the bug**
Attempting to freshly deploy sumologic-kubernetes-collection helm chart v1.3.3 to v1.18.8 kubernetes with PSP enabled. Setup pod fails to schedule due to 'Error: container has runAsNonRoot and image has non-numeric user (fluent), cannot verify user is non-root'

**Logs**
```
Events:
  Type     Reason     Age                    From               Message
  ----     ------     ----                   ----               -------
  Normal   Scheduled  29m                    default-scheduler  Successfully assigned sumologic/sumologic-sumologic-setup-9gwvx to server24
  Normal   Pulling    29m                    kubelet            Pulling image "public.ecr.aws/u5z5f8z6/kubernetes-fluentd:1.3.3"
  Normal   Pulled     29m                    kubelet            Successfully pulled image "public.ecr.aws/u5z5f8z6/kubernetes-fluentd:1.3.3"
  Warning  Failed     27m (x12 over 29m)     kubelet            Error: container has runAsNonRoot and image has non-numeric user (fluent), cannot verify user is non-root
  Normal   Pulled     4m50s (x116 over 29m)  kubelet            Container image "public.ecr.aws/u5z5f8z6/kubernetes-fluentd:1.3.3" already present on machine
```

**Command used to install/upgrade Collection**
```
helmfile -e clustername -l name=sumologic sync --set sumologic.accessId=12356 --set sumologic.accessKey=ABCDEF --set sumologic.setupEnabled=true
```

**Configuration**
```
image:
  tag: {{ index .Values.image.tag "sumologic" }}

sumologic:
  setupEnabled: false
  clusterName: {{ .Environment.Name }}
  metrics:
    enabled: false

fluentd:
  podSecurityPolicy:
    create: true
  logs:
    statefulset:
      priorityClassName: base-services
    autoscaling:
      enabled: true
      minReplicas: 3
      maxReplicas: 10
      targetCPUUtilizationPercentage: 50
  metrics:
    enabled: false
  events:
    statefulset:
      priorityClassName: base-services

fluent-bit:
  resources:
    limits:
      cpu: 100m
      memory: 128Mi
    requests:
      cpu: 10m
      memory: 8Mi

prometheus-operator:
  prometheusOperator:
    enabled: false
  prometheus-node-exporter:
    service:
      port: 9200
      targetPort: 9200

```

**Environment (please complete the following information):**
 - **Collection version** v1.3.3
 - **Kubernetes version** v1.18.8
 - **Cloud provider**: N/A
 - **Others**:

**Anything else do we need to know**
podsecuritypolicies are enabled

**Comment by pmalek-sumo** [comment]
Hi @ag237,

In order to fix your issue you can edit `securiyContext` for your `fluentd` pods by adding `runAsUser: 999`

```yaml
fluentd:
  ## Specifies whether a PodSecurityPolicy should be created
  podSecurityPolicy:
    create: true
  securityContext:
    ## The group ID of all processes in the statefulset containers. By default this needs to be fluent(999).
    fsGroup: 999
    runAsUser: 999
    ...
```

**Comment by ag237** [comment]
So dug into this further. It's the setup container that is failing to schedule:

```
NAME                              READY   STATUS                       RESTARTS   AGE
sumologic-sumologic-setup-tzjgz   0/1     CreateContainerConfigError   0          116s
```

I have a 'kubespray' deployed cluster with the default PSP's enabled. The default 'restricted' PSP requires that pods run as non-root:

```
spec:
...

  runAsUser:
    rule: MustRunAsNonRoot

...
```

Looks like if we can add the same securityContext to the setup-job.yaml, it would work.

**Comment by ag237** [comment]
Looks like this was already added https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/main/deploy/helm/sumologic/templates/setup/setup-job.yaml

**Comment by pmalek-sumo** [comment]
This has already landed in `release-v1.3` branch and will be released with `1.3.5` as soon as it's out.

https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/0a28d013b49c6658be6c3dbf6c25a300fe961136/deploy%2Fhelm%2Fsumologic%2Ftemplates%2Fsetup%2Fsetup-job.yaml#L70-L71

**Comment by pmalek-sumo** [comment]
@ag237 `1.3.5` is out - https://github.com/SumoLogic/sumologic-kubernetes-collection/releases/tag/v1.3.5

I'm closing this for now. Please comment if you're still having issues with this so that we can work this out.
